server/sonar-server/src/main/java/org/sonar/server/issue/filter/RegisterIssueFilters.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * SonarQube, open source software quality management tool.
+ * Copyright (C) 2008-2014 SonarSource
+ * mailto:contact AT sonarsource DOT com
+ *
+ * SonarQube is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * SonarQube is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.issue.filter;
+
+import org.picocontainer.Startable;
+import org.sonar.api.utils.System2;
+import org.sonar.core.issue.DefaultIssueFilter;
+import org.sonar.core.template.LoadedTemplateDao;
+import org.sonar.core.template.LoadedTemplateDto;
+
+import java.util.Date;
+
+public class RegisterIssueFilters implements Startable {
+
+  static final String BUILTIN_ISSUE_FILTER_UNRESOLVED = "Unresolved Issues";
+  static final String BUILTIN_ISSUE_FILTER_HIDDEN_DEBT = "False Positive and Won't Fix Issues";
+  static final String BUILTIN_ISSUE_FILTER_MY_UNRESOLVED = "My Unresolved Issues";
+
+  private final IssueFilterService issueFilterService;
+
+  private final LoadedTemplateDao loadedTemplateDao;
+
+  private final System2 system;
+
+  public RegisterIssueFilters(IssueFilterService issueFilterService, LoadedTemplateDao loadedTemplateDao, System2 system) {
+    this.issueFilterService = issueFilterService;
+    this.loadedTemplateDao = loadedTemplateDao;
+    this.system = system;
+  }
+
+  @Override
+  public void start() {
+    if (shouldRegisterBuiltinIssueFilters()) {
+      createBuiltinIssueFilters();
+      registerBuiltinIssueFilters();
+    }
+  }
+
+  @Override
+  public void stop() {
+    // do nothing
+  }
+
+  private boolean shouldRegisterBuiltinIssueFilters() {
+    return loadedTemplateDao.countByTypeAndKey(LoadedTemplateDto.ISSUE_FILTER_TYPE, BUILTIN_ISSUE_FILTER_UNRESOLVED) == 0;
+  }
+
+  private void createBuiltinIssueFilters() {
+    Date now = system.newDate();
+
+    DefaultIssueFilter unresolvedIssueFilter = new DefaultIssueFilter().setName(BUILTIN_ISSUE_FILTER_UNRESOLVED)
+      .setShared(true)
+      .setCreatedAt(now)
+      .setUpdatedAt(now)
+      .setData("resolved=false");
+    issueFilterService.save(unresolvedIssueFilter);
```

Comment:
Why not use directly the DAO ? This method is only converting the DefaultIssueFilter in IssueFilterDto. 
Moreover, I think that this DefaultIssueFilter makes no more sense, we should replace it in the futur by the DTO.